tests/test_histogram.py

```diff
@@ -667,6 +667,14 @@ def test_pickle_1():
     assert a == b
 
 
+def test_fill_bool_not_bool():
+    a = bh.Histogram(bh.axis.Boolean())
+
+    a.fill([0, 1, 1, 7, -3])
+
+    assert_array_equal(a.view(), [1, 2])
```

Comment:
@HDembinski Actually, what should this do for non-zero values? Should this follow Boolean logic, and non-zero is True, or should we make it just like `Integer(0, 2)`, as done here? I'm leaning toward the former, as it's consistent with other booleans. So the result here would be `[1, 4]` if I made that change. I didn't have this test on the other axes, but I expect it would have produced [1, 4] or would have failed with a type error.

Reply:
I just checked the current develop, this test produces `[1, 4]`. So will change this PR to match. That means there was (and still will be) an advantage to using `Boolean` over `Integer(0,2, flow...)`